crowdin/src/test/java/com/crowdin/platform/MappingRepositoryTest.kt

```diff
@@ -253,6 +253,116 @@ class MappingRepositoryTest {
         assertThat(actualLanguageInfo, equalTo(null))
     }
 
+    @Test
+    fun validateFilePath_languageMappingTest() {
+        // Given
+        val mappingRepository = givenMappingRepository()
+        val givenLanguageInfo = givenLanguageInfo()
+        val expectedPath = "/ua/strings.xml"
+
+        // When
+        val givenFilePathWithTwoLettersPattern = "/%two_letters_code%/strings.xml"
+        val givenFilePathWithThreeLettersPattern = "/%two_letters_code%/strings.xml"
+        val givenFormattedCode = "uk"
+        val givenLanguageMapping = givenLanguageMapping()
+
+        // Then Two letters pattern
+        assertThat(
+            mappingRepository.validateFilePath(
```

Comment:
and this is what we test `When` part:

```
mappingRepository.validateFilePath(
                givenFilePathWithThreeLettersPattern,
                givenLanguageInfo,
                givenFormattedCode,
                givenLanguageMapping
            )
```

this method call can be stored in `val` property and later validated in `Then` block with assert.

Same for methods below.

Reply:
resolved